packages/maskbook/src/plugins/Trader/apis/trending/hotfix.ts

```diff
@@ -45,25 +43,27 @@ const ID_ADDRESS_MAP: {
     [DataProvider.COIN_GECKO]: {
         'crust-network': '0x32a7c02e79c4ea1008dd6564b35f131428673c41',
     },
+    [DataProvider.UNISWAP]: {},
 }
 
 export function resolveAlias(keyword: string, dataProvider: DataProvider) {
     if (dataProvider === DataProvider.COIN_MARKET_CAP)
         return KEYWORD_ALIAS_MAP[DataProvider.COIN_MARKET_CAP][keyword.toUpperCase()] ?? keyword
     if (dataProvider === DataProvider.COIN_GECKO)
         return KEYWORD_ALIAS_MAP[DataProvider.COIN_GECKO][keyword.toUpperCase()] ?? keyword
-    unreachable(dataProvider)
+    if (dataProvider === DataProvider.UNISWAP) return keyword
+    unreachable(dataProvider as never)
 }
 
 export function resolveCoinId(keyword: string, dataProvider: DataProvider) {
     if (dataProvider === DataProvider.COIN_MARKET_CAP)
         return KEYWORK_ID_MAP[DataProvider.COIN_MARKET_CAP][keyword.toUpperCase()]
     if (dataProvider === DataProvider.COIN_GECKO) return KEYWORK_ID_MAP[DataProvider.COIN_GECKO][keyword.toUpperCase()]
-    unreachable(dataProvider)
+    unreachable(dataProvider as never)
 }
 
 export function resolveCoinAddress(id: string, dataProvider: DataProvider) {
     if (dataProvider === DataProvider.COIN_MARKET_CAP) return ID_ADDRESS_MAP[DataProvider.COIN_MARKET_CAP][id]
     if (dataProvider === DataProvider.COIN_GECKO) return ID_ADDRESS_MAP[DataProvider.COIN_GECKO][id]
-    unreachable(dataProvider)
+    unreachable(dataProvider as never)
```

Comment:
Should not use `as never` here.

Reply:
why?

Reply:
Force devs to impl this function for future data providers.

Reply:
Impl which function, resolveCoinAddress?

Reply:
yes.